module/VuFind/src/VuFind/Controller/AbstractSearch.php

```diff
@@ -430,6 +439,10 @@ protected function getSearchResultsView($setupCallback = null)
                 $this->flashMessenger()->addErrorMessage($error);
             }
         }
+        // For the header link
+        $this->layout()->setVariable('searchOrigin', $searchOrigin);
+        // For the records URL
+        $view->setVariable('searchOrigin', $searchOrigin);
```

Comment:
For consistency of style, I think you could use the shorter syntax here:

```
        // For the header link
        $this->layout()->searchOrigin = $searchOrigin;
        // For the records URL
        $view->searchOrigin = $searchOrigin;
```

...though I guess if the searchOrigin is always set in the layout, it might make sense to always read it from there instead of setting it in two places. Is there a strong reason not to?